### ISSUE TITLE
[UX] Out of The Box experience initiative

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#2946]
In DrupalCon Dublin 2016, there was this session by [Angie Byron](http://webchick.net/about) titled "[How our competitors are kicking Drupal's ass (and what we can do about it)](https://events.drupal.org/dublin2016/sessions/how-our-competitors-are-kicking-drupals-ass-and-what-we-can-do-about-it)". Now, there's and official initiative in d.org to deal about it:

[Out of The Box experience initiative](https://www.drupal.org/project/ideas/issues/2847582)

The goal is to add sample content presented in a well designed theme, presented as a food magazine. Using recipes and feature articles this example site will make Drupal look much better right from the start and help evaluators explore core Drupal concepts like content types, fields, blocks, views, taxonomy, etc.

Together this will also be a useful tool for agencies to demo what Drupal can do for their (potential) clients.

![umami](https://user-images.githubusercontent.com/2423362/35121340-29e7bc8e-fcef-11e7-8eeb-9f032ae0bfaa.jpg)

> **Installing Drupal currently gives you a mostly empty box. There is no demo content nor much preconfigured functionality. This makes for a weak first impression and makes it hard for the evaluator to connect the dots and find out what Drupal can do for them.**

This ticket here is to discuss if this is something that we might want to do in Backdrop as well.

**Comment by tomgrandy** [comment]
This is something that has helped me figure out how a CMS works by installing a theme with content and then walking it back to see how it was made.  It isn't so much an install that I would use for a project of my own, but it certainly helps me show others what Backdrop is capable of without having to build out an entire site of my own.  

Granted, WE know that you can build just about anything in Backdrop but others need to be able to see that easily with a sample site.  As Jen mentioned, we could do a sandbox site that cleans itself up every day for people to play with, but it is vital that whatever site showcases some of the best features of Backdrop and its ease of use.

Definitely interested in helping out with this initiative for I believe that it would be HUGE to help people become familiar with the CMS and realize how far Backdrop has come in the past three years.

**Comment by drupol** [comment]
Following the issue :-)

**Comment by tomgrandy** [comment]
Looks like the Out of the Box Experience has been adopted to Drupal 8.6.x core.

https://mark.ie/blog/web-development/out-box-has-just-been-committed-drupal-core-86x?utm_source=drupal-newsletter&utm_medium=email&utm_campaign=drupal-newsletter-20180125

**Comment by cellear** [comment]
In the design/ux meeting on Thursday we discussed three ideas for sample data we could include with Backdrop:

1. Recipes
2. Dragons
3. Pets

#1 is basically just copying Drupal's OOTB initiative.  This has the advantage of being proven and community-approved.  Additionally, the materials are easily available with a compatible license.  The obvious disadvantage is that it copies Drupal a little too closely; we'd like to be more independent than that, right?

#2 Dragons are fun, we have a dragon as our mascot, there are lots of interesting images and stories related to dragons.  The biggest drawback I see with them is that a lot of the most interesting dragons have copyright issues.

#3 Pets.  Since most people use the Internet mostly to watch cat videos anyway, people will relate to content about animals, and can click around learning about Backdrop while they like at the cute kitties and puppies.  And, perhaps, horses -- or even dragons.

I'm going to start prototyping a pets site.

This is getting slighting ahead of myself (I need to put together a site first) but what are options for packaging up content?  (It's always been a complaint of mine with Drupal -- I long for something like Wordpress's "export" functionality.)  @jenlampton suggested that it may be desirable for the sample content to be a separate module, so that we don't bloat the core codebase.

**Comment by jenlampton** [comment]
> what are options for packaging up content?

I'd recommend syncing config files for the content type, fields, views, and layouts. All machine names probably should be prefixed with `example_` for easy deletion later.

For the content itself, we have several options:
* We could require devel generate and create it on the fly.
* We could require feeds and pull it in from a CSV file, or similar.
* We could create a structured array of data, and use an old fashioned update (or install) hook. Since we know the field names, this would be pretty straightforward with `$node->save()`.

**Comment by klonos** [comment]
> We could require devel generate and create it on the fly.

That means Devel in core though. Right? ...not that I am objecting to this 😄. We already have a `Configuration` >> `Development` menu entry and a settings page to go with it, and also the "stray" `Enable views performance statistics/debug messages via the Devel module` setting in `/admin/structure/views/settings/advanced` 😉.

**Comment by jenlampton** [comment]
> That means Devel in core though. Right?

No, Not if the "sample content" is a contrib module that appears at the top of the project browser.

**Comment by klonos** [comment]
Cross-linking #1591 #3126 and the very old #99 😉
